haystack/utils/augment_squad.py

```diff
@@ -0,0 +1,235 @@
+    transformers_model.to(device)
+    transformers_tokenizer = AutoTokenizer.from_pretrained(tokenizer, use_fast=False)
+    # load glove for words that do not have one distinct token, but are split into subwords
+    word_id_mapping, id_word_mapping, vectors = load_glove(glove_path=glove_path, device=device)
+
+    # load squad dataset
+    with open(squad_path, "r") as f:
+        squad = json.load(f)
+        
+    topics = []
+
+    for topic in tqdm(squad["data"]):
+        paragraphs = []
+        for paragraph in topic["paragraphs"]:
+            # make every question unanswerable as answer strings will probably match and aren't relevant for distillation
+            for question in paragraph["qas"]:
+                question["answers"] = []
+                question["is_impossible"] = True
+            context = paragraph["context"]
+            contexts = augment(word_id_mapping=word_id_mapping, id_word_mapping=id_word_mapping, vectors=vectors,
+                model=transformers_model, tokenizer=transformers_tokenizer, text=context, multiplication_factor=multiplication_factor,
+                word_possibilities=word_possibilities, replace_probability=replace_probability, device=device, batch_size=batch_size)
+            paragraphs_ = []
+            for context in contexts:
+                new_paragraph = deepcopy(paragraph)
+                new_paragraph["context"] = context
+                paragraphs_.append(new_paragraph)
+            paragraphs += paragraphs_
+        topic["paragraphs"] = paragraphs
+        topics.append(topic)
+    squad["topics"] = topics
+
+    # save new dataset
+    with open(output_path, "w") as f:
+        json.dump(squad, f)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--squad_path", type=Path, required=True, help="Path to the squad json file")
+    parser.add_argument("--output_path", type=Path, required=True, help="Path to save augmented dataaset")
+    parser.add_argument("--multiplication_factor", type=int, default=20, help="Factor by which dataset size is multiplied")
+    parser.add_argument("--word_possibilities", type=int, default=5, help="Number of possible words to choose from when replacing a word")
+    parser.add_argument("--replace_probability", type=float, default=0.4, help="Probability of replacing a word")
+    parser.add_argument("--glove_path", type=Path, default="glove.txt", help="Path to the glove file")
+    parser.add_argument("--batch_size", type=int, default=16, help="Batch size for forward pass")
+    parser.add_argument("--device", type=str, default="cuda:0", help="Device to use")
+    parser.add_argument("--model", type=str, default="bert-base-uncased", help="Huggingface model identifier for MLM model")
+    parser.add_argument("--tokenizer", type=str, default="bert-base-uncased", help="Huggingface tokenizer identifier for MLM model")
+
+    args = parser.parse_args()
+
+    augment_squad(**vars(args))
```

Comment:
Regarding testing: I would suggest that for a test we load a small/tiny squad file with [SquadData](https://github.com/deepset-ai/haystack/blob/13510aa753d4e390f244398cc50654185dddbcde/haystack/utils/squad_data.py#L23) and count the number of questions with https://github.com/deepset-ai/haystack/blob/13510aa753d4e390f244398cc50654185dddbcde/haystack/utils/squad_data.py#L146
Next step is to run the `augment_squad()` method and in the end load the result again with SquadData and count again to see whether the size of the dataset was multiplied as expected by `multiplication_factor`. What do you think?